# Best Sklearn Models

In [1]:
import numpy
import csv
import time
import pandas
import signal
import warnings
warnings.filterwarnings("ignore")

from sklearn.metrics import make_scorer
from sklearn.model_selection import GridSearchCV
from sklearn.model_selection import cross_validate
from sklearn.pipeline import Pipeline

from sklearn.multioutput import ClassifierChain
from sklearn.multioutput import MultiOutputRegressor
from sklearn.multioutput import MultiOutputClassifier

from sklearn.ensemble.weight_boosting import AdaBoostClassifier
from sklearn.ensemble.weight_boosting import AdaBoostRegressor
from sklearn.ensemble.bagging import BaggingClassifier
from sklearn.ensemble.bagging import BaggingRegressor
from sklearn.ensemble.forest import ExtraTreesClassifier
from sklearn.ensemble.forest import ExtraTreesRegressor
from sklearn.ensemble.gradient_boosting import GradientBoostingClassifier
from sklearn.ensemble.gradient_boosting import GradientBoostingRegressor
from sklearn.ensemble.forest import RandomForestClassifier
from sklearn.ensemble.forest import RandomForestRegressor

from sklearn.linear_model.bayes import ARDRegression
from sklearn.linear_model.bayes import BayesianRidge
from sklearn.naive_bayes import BernoulliNB
from sklearn.cross_decomposition.cca_ import CCA
from sklearn.tree.tree import DecisionTreeClassifier
from sklearn.tree.tree import DecisionTreeRegressor
from sklearn.linear_model.coordinate_descent import ElasticNet
from sklearn.tree.tree import ExtraTreeClassifier
from sklearn.tree.tree import ExtraTreeRegressor
from sklearn.naive_bayes import GaussianNB
from sklearn.gaussian_process.gpc import GaussianProcessClassifier
from sklearn.gaussian_process.gpr import GaussianProcessRegressor
from sklearn.linear_model.huber import HuberRegressor
from sklearn.neighbors.classification import KNeighborsClassifier
from sklearn.neighbors.regression import KNeighborsRegressor
from sklearn.kernel_ridge import KernelRidge
from sklearn.semi_supervised.label_propagation import LabelPropagation
from sklearn.semi_supervised.label_propagation import LabelSpreading
from sklearn.linear_model.least_angle import Lars
from sklearn.linear_model.coordinate_descent import Lasso
from sklearn.linear_model.least_angle import LassoLars
from sklearn.linear_model.least_angle import LassoLarsIC
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
from sklearn.linear_model.base import LinearRegression
from sklearn.svm.classes import LinearSVC
from sklearn.svm.classes import LinearSVR
from sklearn.linear_model.logistic import LogisticRegression
from sklearn.neural_network.multilayer_perceptron import MLPClassifier
from sklearn.neural_network.multilayer_perceptron import MLPRegressor
from sklearn.linear_model.coordinate_descent import MultiTaskElasticNet
from sklearn.linear_model.coordinate_descent import MultiTaskLasso
from sklearn.naive_bayes import MultinomialNB
from sklearn.neighbors.nearest_centroid import NearestCentroid
from sklearn.svm.classes import NuSVC
from sklearn.svm.classes import NuSVR
from sklearn.linear_model.omp import OrthogonalMatchingPursuit
from sklearn.cross_decomposition.pls_ import PLSCanonical
from sklearn.cross_decomposition.pls_ import PLSRegression
from sklearn.linear_model.passive_aggressive import PassiveAggressiveClassifier
from sklearn.linear_model.passive_aggressive import PassiveAggressiveRegressor
from sklearn.linear_model.perceptron import Perceptron
from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
from sklearn.linear_model.ransac import RANSACRegressor
from sklearn.neighbors.classification import RadiusNeighborsClassifier
from sklearn.neighbors.regression import RadiusNeighborsRegressor
from sklearn.linear_model.ridge import Ridge
from sklearn.linear_model.ridge import RidgeClassifier
from sklearn.linear_model.stochastic_gradient import SGDClassifier
from sklearn.linear_model.stochastic_gradient import SGDRegressor
from sklearn.svm.classes import SVC
from sklearn.svm.classes import SVR
from sklearn.linear_model.theil_sen import TheilSenRegressor

In [2]:
psms = pandas.read_csv('11 PS MS.csv', delimiter=',', encoding='latin-1').fillna('')
len(psms)

238

In [3]:
ptmb = pandas.read_csv('11 PT MB.csv', delimiter=',', encoding='latin-1').fillna('')
ptme = pandas.read_csv('12 PT ME.csv', delimiter=',', encoding='latin-1').fillna('')
ptmeb = pandas.read_csv('13 PT MEB.csv', delimiter=',', encoding='latin-1').fillna('')
len(ptmb) + len(ptme) + len(ptmeb)

70

In [4]:
cfmb = pandas.read_csv('14 CF MB.csv', delimiter=',', encoding='latin-1').fillna('')
cfme = pandas.read_csv('15 CF ME.csv', delimiter=',', encoding='latin-1').fillna('')
cfmeb = pandas.read_csv('16 CF MEB.csv', delimiter=',', encoding='latin-1').fillna('')
len(cfmb) + len(cfme) + len(cfmeb)

70

In [5]:
osmb = pandas.read_csv('17 OS MB.csv', delimiter=',', encoding='latin-1').fillna('')
osme = pandas.read_csv('18 OS ME.csv', delimiter=',', encoding='latin-1').fillna('')
osmeb = pandas.read_csv('19 OS MEB.csv', delimiter=',', encoding='latin-1').fillna('')
len(ptmb) + len(ptme) + len(ptmeb)

70

In [25]:
def lowest_correct(trues, preds):
    num_of_options = len(trues)
    drop_down_options = list(reversed(numpy.argsort(preds)))
    correct_options = [i for i in range(num_of_options) if trues[i]==1]
    return max([drop_down_options.index(correct_option) for correct_option in correct_options]) + 1
def average_lowest_correct(list_of_trues, list_of_preds):
    length = len(list_of_trues)
    return numpy.mean([lowest_correct(list(list_of_trues.iloc[i]), list(list_of_preds[i])) for i in range(length)])
def cv_ensemble(model, xs, ys):
    temp = cross_validate(model, xs, ys, return_train_score=True, scoring=make_scorer(average_lowest_correct), n_jobs=1, cv=10, verbose=2)
    return numpy.mean(temp['test_score']).round(2)

In [26]:
rdg = MultiOutputRegressor(AdaBoostRegressor(Ridge(alpha=10, fit_intercept=True, max_iter=3000, normalize=False, solver='saga')))
lsv = MultiOutputRegressor(AdaBoostRegressor(LinearSVR(C=0.2, dual=False, epsilon=0.2, fit_intercept=True, loss='squared_epsilon_insensitive', max_iter=1000, tol=0.01)))
pag = MultiOutputRegressor(AdaBoostRegressor(PassiveAggressiveRegressor(C=0.1, average=True, epsilon=0.1, fit_intercept=True, max_iter=4, loss='squared_epsilon_insensitive', tol=0.00001)))
rfr = MultiOutputRegressor(AdaBoostRegressor(RandomForestRegressor(max_features='log2', bootstrap=True, oob_score=True, min_samples_leaf=2, min_samples_split=2, min_weight_fraction_leaf=0)))
mdls = [rdg, lsv, pag, rfr]
mdls = [lsv, pag, rfr]

## Process Step

In [27]:
tfidf = pandas.read_csv('07 NM PS TFIDF.csv', delimiter=',', encoding='latin-1').fillna('')
tfidf[:3]

_____________________  _____________________ follow  able  acceptable  \
0                    0.0                           0.0   0.0         0.0   
1                    0.0                           0.0   0.0         0.0   
2                    0.0                           0.0   0.0         0.0   

   accessory  accessory require  accidentally  accompany  accord  account  \
0        0.0                0.0           0.0        0.0     0.0      0.0   
1        0.0                0.0           0.0        0.0     0.0      0.0   
2        0.0                0.0           0.0        0.0     0.0      0.0   

    ...    wrong ct  wrong info  wrong info mri  wrong patient  wrong pt  \
0   ...         0.0         0.0             0.0            0.0       0.0   
1   ...         0.0         0.0             0.0            0.0       0.0   
2   ...         0.0         0.0             0.0            0.0       0.0   

   wrong set  wrong tattoo  xrt  xrt date  xrt pt  
0        0.0           0.0  0.0       0.0     0.0  
1        0.0           0.0  0.0       0.0     0.0  
2        0.0           0.0  0.0       0.0     0.0  

[3 rows x 2261 columns]

In [28]:
ohe = pandas.read_csv('07 NM PS OHE.csv', delimiter=',', encoding='latin-1').fillna('')
ohe[:3]

Treatment delivery  Radiation treatment prescription scheduling  \
0                   1                                            0   
1                   0                                            0   
2                   0                                            0   

   Pre-treatment quality assurance  Post-treatment completion  \
0                                0                          0   
1                                0                          1   
2                                1                          0   

   Interventional procedure for planning and/or delivery  \
0                                                  0       
1                                                  0       
2                                                  0       

   Contouring and planning  Imaging for treatment planning  \
0                        0                               0   
1                        0                               0   
2                        0                               0   

   Patient medical consultation and physician assessment  \
0                                                  0       
1                                                  0       
2                                                  0       

   On-treatment quality assurance  
0                               0  
1                               0  
2                               0

In [29]:
for mdl in mdls:
    print(cv_ensemble(mdl, tfidf, ohe), mdl)

[CV]  ................................................................
[CV] ................................................. , total=  16.0s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   16.8s remaining:    0.0s


[CV] ................................................. , total=  17.2s
[CV]  ................................................................
[CV] ................................................. , total=  11.8s
[CV]  ................................................................
[CV] ................................................. , total=  11.8s
[CV]  ................................................................
[CV] ................................................. , total=  15.9s
[CV]  ................................................................
[CV] ................................................. , total=  13.6s
[CV]  ................................................................
[CV] ................................................. , total=  14.6s
[CV]  ................................................................
[CV] ................................................. , total=  18.5s
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed:  2.6min finished


[CV] ................................................. , total=  25.2s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   25.9s remaining:    0.0s


[CV] ................................................. , total=  28.5s
[CV]  ................................................................
[CV] ................................................. , total=  25.7s
[CV]  ................................................................
[CV] ................................................. , total=  29.8s
[CV]  ................................................................
[CV] ................................................. , total=  28.8s
[CV]  ................................................................
[CV] ................................................. , total=  27.1s
[CV]  ................................................................
[CV] ................................................. , total=  32.4s
[CV]  ................................................................
[CV] ................................................. , total=  28.1s
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed:  4.8min finished


[CV] ................................................. , total=  45.7s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   47.7s remaining:    0.0s


[CV] ................................................. , total=  58.3s
[CV]  ................................................................
[CV] ................................................. , total=  53.8s
[CV]  ................................................................
[CV] ................................................. , total=  59.1s
[CV]  ................................................................
[CV] ................................................. , total=  58.3s
[CV]  ................................................................
[CV] ................................................. , total= 1.1min
[CV]  ................................................................
[CV] ................................................. , total= 1.0min
[CV]  ................................................................
[CV] ................................................. , total= 1.1min
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed: 10.2min finished


### Default
1.69, MultiOutput(Ridge)
### Tuned
1.67 MultiOutput(BaggingRegressor(LinearSVR)) (k=2)

1.67 MultiOutput(BaggingRegressor(PassiveAggressiveRegressor)) (k=2)

1.57 MultiOutput(BaggingRegressor(LinearSVR)) (k=10)

## Problem Type

In [33]:
tfidf = pandas.read_csv('07 NM PT TFIDF.csv', delimiter=',', encoding='latin-1').fillna('')
tfidf[:3]

____________________  _____________________  _____________________ follow  \
0                   0.0                    0.0                           0.0   
1                   0.0                    0.0                           0.0   
2                   0.0                    0.0                           0.0   

   able  acceptable  accessory  accessory require  accidentally  accommodate  \
0   0.0         0.0        0.0                0.0           0.0          0.0   
1   0.0         0.0        0.0                0.0           0.0          0.0   
2   0.0         0.0        0.0                0.0           0.0          0.0   

   accompany   ...    work  would  write  writer  wrong  wrong patient  \
0        0.0   ...     0.0    0.0    0.0     0.0    0.0            0.0   
1        0.0   ...     0.0    0.0    0.0     0.0    0.0            0.0   
2        0.0   ...     0.0    0.0    0.0     0.0    0.0            0.0   

   wrong tattoo  xrt  xrt date  xrt pt  
0           0.0  0.0       0.0     0.0  
1           0.0  0.0       0.0     0.0  
2           0.0  0.0       0.0     0.0  

[3 rows x 1891 columns]

In [34]:
ohe = pandas.read_csv('07 NM PT OHE.csv', delimiter=',', encoding='latin-1').fillna('')
ohe[:3]

Wrong target or OAR contours  \
0                             0   
1                             0   
2                             0   

   Treatment plan acceptable but not physically deliverable  \
0                                                  0          
1                                                  0          
2                                                  0          

   Excess imaging dose   \
0                     0   
1                     0   
2                     0   

   Failure to perform on-treatment imaging as per instructions  Wrong patient  \
0                                                  0                        0   
1                                                  0                        0   
2                                                  0                        0   

   Fall or other patient injury or medical condition  \
0                                                  0   
1                                                  0   
2                                                  0   

   Wrong anatomical site (excluding laterality)  \
0                                             0   
1                                             0   
2                                             0   

   Treatment plan (isodose distribution) unacceptable  \
0                                                  0    
1                                                  0    
2                                                  0    

   Systematic hardware/software (including dose-volume) error  \
0                                                  0            
1                                                  0            
2                                                  0            

   Interventional procedure error (Retired value)        ...          \
0                                               0        ...           
1                                               0        ...           
2                                               0        ...           

   Inappropriate or poorly informed decision to treat or plan  \
0                                                  0            
1                                                  0            
2                                                  0            

   Inadequate coordination of combined modality care  \
0                                                  0   
1                                                  0   
2                                                  0   

   Wrong, missing, mislabeled, or damaged treatment accessories  \
0                                                  1              
1                                                  0              
2                                                  0              

   Wrong side (laterality)  \
0                        0   
1                        0   
2                        0   

   Wrong prescription dose-fractionation or calculation error  \
0                                                  0            
1                                                  0            
2                                                  0            

   Radiation therapy scheduling error  \
0                                   0   
1                                   1   
2                                   1   

   Treatment not delivered - personnel/hardware/software failure  \
0                                                  0               
1                                                  0               
2                                                  0               

   Wrong patient position, setup point, or shift  Wrong planning margins  \
0                                              0                       0   
1                                              0                       0   
2                                              0                       0   

   Allergic reaction  
0                  0  
1                  0  
2                

In [32]:
JJJJJJ

[CV]  ................................................................
[CV] ................................................. , total=  21.4s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   22.6s remaining:    0.0s


[CV] ................................................. , total=  26.7s
[CV]  ................................................................
[CV] ................................................. , total=  25.2s
[CV]  ................................................................
[CV] ................................................. , total=  25.5s
[CV]  ................................................................
[CV] ................................................. , total=  26.2s
[CV]  ................................................................
[CV] ................................................. , total=  26.9s
[CV]  ................................................................
[CV] ................................................. , total=  25.3s
[CV]  ................................................................
[CV] ................................................. , total=  24.7s
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed:  4.4min finished


[CV] ................................................. , total=  39.3s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   40.3s remaining:    0.0s


[CV] ................................................. , total=  40.3s
[CV]  ................................................................
[CV] ................................................. , total=  36.2s
[CV]  ................................................................
[CV] ................................................. , total=  36.7s
[CV]  ................................................................
[CV] ................................................. , total=  36.3s
[CV]  ................................................................
[CV] ................................................ , total=163.1min
[CV]  ................................................................
[CV] ................................................. , total=  32.5s
[CV]  ................................................................
[CV] ................................................. , total=  35.8s
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed: 168.8min finished


[CV] ................................................. , total= 1.1min
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:  1.1min remaining:    0.0s


[CV] ................................................. , total= 1.1min
[CV]  ................................................................
[CV] ................................................. , total= 1.1min
[CV]  ................................................................
[CV] ................................................. , total=72.8min
[CV]  ................................................................
[CV] ................................................. , total= 1.1min
[CV]  ................................................................


KeyboardInterrupt: 

In [36]:
rdg = MultiOutputRegressor(AdaBoostRegressor(Ridge(alpha=10, fit_intercept=True, max_iter=3000, normalize=False, solver='saga')))
lsv = MultiOutputRegressor(AdaBoostRegressor(LinearSVR(C=0.2, dual=False, epsilon=0.2, fit_intercept=True, loss='squared_epsilon_insensitive', max_iter=1000, tol=0.01)))
pag = MultiOutputRegressor(AdaBoostRegressor(PassiveAggressiveRegressor(C=0.1, average=True, epsilon=0.1, fit_intercept=True, max_iter=4, loss='squared_epsilon_insensitive', tol=0.00001)))
rfr = MultiOutputRegressor(AdaBoostRegressor(RandomForestRegressor(max_features='log2', bootstrap=True, oob_score=True, min_samples_leaf=2, min_samples_split=2, min_weight_fraction_leaf=0)))
mdls = [rdg, lsv, pag, rfr]
mdls = [rfr]

In [37]:
for mdl in mdls:
    print(cv_ensemble(mdl, tfidf, ohe), mdl)

[CV]  ................................................................
[CV] ................................................. , total=  59.5s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:  1.0min remaining:    0.0s


[CV] ................................................. , total=  56.0s
[CV]  ................................................................
[CV] ................................................. , total=  55.7s
[CV]  ................................................................
[CV] ................................................. , total=  53.3s
[CV]  ................................................................
[CV] ................................................. , total= 1.0min
[CV]  ................................................................
[CV] ................................................. , total= 1.0min
[CV]  ................................................................
[CV] ................................................. , total= 1.1min
[CV]  ................................................................
[CV] ................................................. , total=  56.1s
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed: 10.3min finished


### Default
2.90 MultiOutputRegressor(BaggingRegressor(Ridge))
### Tuned
2.74 MultiOutputRegressor(BaggingRegressor(PassiveAggressiveRegressor)) (k=2)

2.36 MultiOutputRegressor(BaggingRegressor(PassiveAggressiveRegressor)) （k=10)

## Contributing Factors

In [38]:
tfidf = pandas.read_csv('07 NM CF TFIDF.csv', delimiter=',', encoding='latin-1').fillna('')
tfidf[:3]

____________________  _____________________  _____________________ follow  \
0                   0.0                    0.0                           0.0   
1                   0.0                    0.0                           0.0   
2                   0.0                    0.0                           0.0   

   able  acceptable  access  accessory  accessory require  accidentally  \
0   0.0         0.0     0.0        0.0                0.0           0.0   
1   0.0         0.0     0.0        0.0                0.0           0.0   
2   0.0         0.0     0.0        0.0                0.0           0.0   

   accommodate   ...    wrong ct  wrong info  wrong info mri  wrong patient  \
0          0.0   ...         0.0         0.0             0.0            0.0   
1          0.0   ...         0.0         0.0             0.0            0.0   
2          0.0   ...         0.0         0.0             0.0            0.0   

   wrong pt  wrong set  wrong tattoo  xrt  xrt date  xrt pt  
0       0.0        0.0           0.0  0.0       0.0     0.0  
1       0.0        0.0           0.0  0.0       0.0     0.0  
2       0.0        0.0           0.0  0.0       0.0     0.0  

[3 rows x 2604 columns]

In [39]:
ohe = pandas.read_csv('07 NM CF OHE.csv', delimiter=',', encoding='latin-1').fillna('')
ohe[:3]

Organizational and/or workspace resources inadequate (excluding human resources)  \
0                                                  0                                  
1                                                  0                                  
2                                                  0                                  

   Unnamed: 1  \
0           1   
1           1   
2           1   

   Patient or family member medical condition preference or behaviour  \
0                                                  0                    
1                                                  0                    
2                                                  0                    

   Equipment quality assurance and/or maintenance inadequate  \
0                                                  0           
1                                                  0           
2                                                  0           

   Handoffs inadequate  \
0                    0   
1                    0   
2                    0   

   Communication or documentation inadequate (patient specific)  \
0                                                  0              
1                                                  0              
2                                                  0              

   Policies and/or procedures non-existent or inadequate  \
0                                                  0       
1                                                  0       
2                                                  0       

   Patient or family member medical condition, preference or behaviour  \
0                                                  0                     
1                                                  0                     
2                                                  0                     

   Policies and/or procedures not followed  \
0                                        0   
1                                        0   
2                                        1   

   Unfamiliar treatment approach or radiation treatment technique  \
0                                                  0                
1                                                  0                
2                                                  0                

               ...               \
0              ...                
1              ...                
2              ...                

   Equipment software or hardware design, including 'human factors' design, inadequate  \
0                                                  0                                     
1                                                  0                                     
2                                                  0                                     

   Failure to identify potential risks  \
0                                    0   
1                                    0   
2                                    0   

   Equipment software or hardware commissioning, calibration or acceptance testing inadequate  \
0                                                  0                                            
1                                                  0                                            
2                                                  0                                            

   Distraction or diversions involving staff  Change management  \
0                                          1                  0   
1                                          0                  0   
2                                          0                  0   

   Human resources inadequate  Expectation bias involving staff  \
0                           0                                 0   
1                           1                                 0   
2                           0                                 0   

   Staff behaviour  Staff education or training inadequate  \
0

In [40]:
rdg = MultiOutputRegressor(AdaBoostRegressor(Ridge(alpha=10, fit_intercept=True, max_iter=3000, normalize=False, solver='saga')))
lsv = MultiOutputRegressor(AdaBoostRegressor(LinearSVR(C=0.2, dual=False, epsilon=0.2, fit_intercept=True, loss='squared_epsilon_insensitive', max_iter=1000, tol=0.01)))
pag = MultiOutputRegressor(AdaBoostRegressor(PassiveAggressiveRegressor(C=0.1, average=True, epsilon=0.1, fit_intercept=True, max_iter=4, loss='squared_epsilon_insensitive', tol=0.00001)))
rfr = MultiOutputRegressor(AdaBoostRegressor(RandomForestRegressor(max_features='log2', bootstrap=True, oob_score=True, min_samples_leaf=2, min_samples_split=2, min_weight_fraction_leaf=0)))
mdls = [lsv, pag, rfr]

In [41]:
for mdl in mdls[1:]:
    print(cv_ensemble(mdl, tfidf, ohe), mdl)

[CV]  ................................................................
[CV] ................................................. , total= 1.2min
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:  1.2min remaining:    0.0s


[CV] ................................................. , total= 1.5min
[CV]  ................................................................
[CV] ................................................. , total= 1.4min
[CV]  ................................................................
[CV] ................................................. , total= 1.4min
[CV]  ................................................................
[CV] ................................................. , total= 1.4min
[CV]  ................................................................
[CV] ................................................. , total= 1.4min
[CV]  ................................................................
[CV] ................................................. , total= 1.4min
[CV]  ................................................................
[CV] ................................................. , total= 1.3min
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed: 13.9min finished


[CV] ................................................. , total= 3.0min
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:  3.1min remaining:    0.0s


[CV] ................................................. , total= 2.9min
[CV]  ................................................................
[CV] ................................................. , total= 2.8min
[CV]  ................................................................
[CV] ................................................. , total= 2.5min
[CV]  ................................................................
[CV] ................................................. , total= 2.8min
[CV]  ................................................................
[CV] ................................................. , total= 2.4min
[CV]  ................................................................
[CV] ................................................. , total= 3.0min
[CV]  ................................................................
[CV] ................................................. , total= 2.7min
[CV]  ................................................................
[CV] .

[Parallel(n_jobs=1)]: Done  10 out of  10 | elapsed: 28.9min finished


## Default

    5.72
    
## Tuned:
    
    5.25 MultiOutputRegressor(BaggingRegressor(RandomForestRegressor))

## Overall Severity

In [22]:
tfidf = pandas.read_csv('07 NM OS TFIDF.csv', delimiter=',', encoding='latin-1').fillna('')
tfidf[:3]

____________________  _____________________  _____________________ follow  \
0                   0.0                    0.0                           0.0   
1                   0.0                    0.0                           0.0   
2                   0.0                    0.0                           0.0   

   able  acceptable  access  accessory  accessory require  accidentally  \
0   0.0         0.0     0.0        0.0                0.0           0.0   
1   0.0         0.0     0.0        0.0                0.0           0.0   
2   0.0         0.0     0.0        0.0                0.0           0.0   

   accommodate   ...    wrong ct  wrong info  wrong info mri  wrong patient  \
0          0.0   ...         0.0         0.0             0.0            0.0   
1          0.0   ...         0.0         0.0             0.0            0.0   
2          0.0   ...         0.0         0.0             0.0            0.0   

   wrong pt  wrong set  wrong tattoo  xrt  xrt date  xrt pt  
0       0.0        0.0           0.0  0.0       0.0     0.0  
1       0.0        0.0           0.0  0.0       0.0     0.0  
2       0.0        0.0           0.0  0.0       0.0     0.0  

[3 rows x 2604 columns]

In [23]:
ohe = pandas.read_csv('07 NM OS OHE.csv', delimiter=',', encoding='latin-1').fillna('')
ohe[:3]

Severe  Moderate  None  Mild
0       0         0     1     0
1       0         0     0     1
2       0         0     1     0

In [24]:
for mdl in mdls:
    print(cv_ensemble(mdl, tfidf, ohe), mdl)

[CV]  ................................................................
[CV] ................................................. , total=   6.8s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:    7.3s remaining:    0.0s


[CV] ................................................. , total=   7.0s
[CV]  ................................................................
[CV] ................................................. , total=   6.9s
1.11 MultiOutputRegressor(estimator=AdaBoostRegressor(base_estimator=LinearSVR(C=0.2, dual=False, epsilon=0.2, fit_intercept=True,
     intercept_scaling=1.0, loss='squared_epsilon_insensitive',
     max_iter=1000, random_state=None, tol=0.01, verbose=0),
         learning_rate=1.0, loss='linear', n_estimators=50,
         random_state=None),
           n_jobs=1)
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   3 out of   3 | elapsed:   22.3s finished


[CV] ................................................. , total=  11.1s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   11.6s remaining:    0.0s


[CV] ................................................. , total=  11.0s
[CV]  ................................................................
[CV] ................................................. , total=  10.7s
1.11 MultiOutputRegressor(estimator=AdaBoostRegressor(base_estimator=PassiveAggressiveRegressor(C=0.1, average=True, epsilon=0.1,
              fit_intercept=True, loss='squared_epsilon_insensitive',
              max_iter=4, n_iter=None, random_state=None, shuffle=True,
              tol=1e-05, verbose=0, warm_start=False),
         learning_rate=1.0, loss='linear', n_estimators=50,
         random_state=None),
           n_jobs=1)
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   3 out of   3 | elapsed:   34.1s finished


[CV] ................................................. , total=  29.9s
[CV]  ................................................................


[Parallel(n_jobs=1)]: Done   1 out of   1 | elapsed:   31.4s remaining:    0.0s


[CV] ................................................. , total=  25.3s
[CV]  ................................................................
[CV] ................................................. , total=  32.9s
1.11 MultiOutputRegressor(estimator=AdaBoostRegressor(base_estimator=RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
           max_features='log2', max_leaf_nodes=None,
           min_impurity_decrease=0.0, min_impurity_split=None,
           min_samples_leaf=2, min_samples_split=2,
           min_wei...tart=False),
         learning_rate=1.0, loss='linear', n_estimators=50,
         random_state=None),
           n_jobs=1)


[Parallel(n_jobs=1)]: Done   3 out of   3 | elapsed:  1.5min finished


## Tuned

1.11 MultiOutputRegressor(BaggingRegressor(LinearSVR))

1.11 MultiOutputRegressor(BaggingRegressor(PassiveAggressiveRegressor))

1.11 MultiOutputRegressor(BaggingRegressor(RandomForestRegressor))